prime-router/src/main/kotlin/credentials/Credential.kt

```diff
@@ -0,0 +1,33 @@
+package gov.cdc.prime.router.credentials
+
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.KotlinModule
+
+// All credential classes must exist in this file to inherit from a sealed class
+// https://kotlinlang.org/docs/reference/sealed-classes.html
+
+data class UserPassCredential(val user: String, val pass: String) : Credential()
+
+@JsonTypeInfo(
+    use = JsonTypeInfo.Id.NAME,
+    include = JsonTypeInfo.As.PROPERTY,
+    property = "@type"
+)
+@JsonSubTypes(
+    JsonSubTypes.Type(value = UserPassCredential::class, name = "UserPass")
+)
```

Comment:
We will need to maintain this list for any additional credential types that get added in the future.